gobblin-runtime/src/main/java/gobblin/runtime/Fork.java

```diff
@@ -134,18 +135,9 @@ public Fork(TaskContext taskContext, Object schema, int branches, int index) thr
       buildWriterIfNotPresent();
     }
 
-    this.recordQueue = BoundedBlockingRecordQueue.newBuilder()
-        .hasCapacity(this.taskState.getPropAsInt(
+    this.recordQueue = new LinkedBlockingQueue<Object>(this.taskState.getPropAsInt(
```

Comment:
Any reason to not use the `BoundedBlockingRecordQueue`, I think we should keep it if we can as it provides important statistics such as queue file ratio, mean rate of put attempts, etc.